Public/Src/Tools/SymbolDaemon/SymbolDaemon.cs

```diff
@@ -0,0 +1,424 @@
+        });
+
+        internal static readonly UriOption ServiceEndpoint = RegisterSymbolConfigOption(new UriOption("service")
+        {
+            ShortName = "s",
+            HelpText = "Symbol service endpoint URI",
+            IsRequired = true,
+        });
+
+        internal static readonly IntOption RetentionDays = RegisterSymbolConfigOption(new IntOption("retentionDays")
+        {
+            ShortName = "rt",
+            HelpText = "Symbol retention time in days",
+            IsRequired = false,
+            DefaultValue = (int)SymbolConfig.DefaultRetention.TotalDays,
+        });
+
+        internal static readonly IntOption HttpSendTimeoutMillis = RegisterSymbolConfigOption(new IntOption("httpSendTimeoutMillis")
+        {
+            HelpText = "Timeout for http requests",
+            IsRequired = false,
+            DefaultValue = (int)SymbolConfig.DefaultHttpSendTimeout.TotalMilliseconds,
+        });
+
+        internal static readonly BoolOption EnableTelemetry = RegisterSymbolConfigOption(new BoolOption("enableTelemetry")
+        {
+            ShortName = "t",
+            HelpText = "Verbose logging",
+            IsRequired = false,
+            DefaultValue = SymbolConfig.DefaultEnableTelemetry,
+        });
+
+        internal static SymbolConfig CreateSymbolConfig(ConfiguredCommand conf)
+        {
+            return new SymbolConfig(
+                requestName: conf.Get(SymbolRequestNameOption),
+                serviceEndpoint: conf.Get(ServiceEndpoint),
+                retention: TimeSpan.FromDays(conf.Get(RetentionDays)),
+                httpSendTimeout: TimeSpan.FromMilliseconds(conf.Get(HttpSendTimeoutMillis)),
+                verbose: conf.Get(Verbose),
+                enableTelemetry: conf.Get(EnableTelemetry),
+                logDir: conf.Get(LogDir));
+        }
+
+        private static Client CreateClient(string serverMoniker, IClientConfig config)
+        {
+            return serverMoniker != null
+                ? Client.Create(serverMoniker, config)
+                : null;
+        }
+
+        internal static readonly Command StartNoServiceCmd = RegisterCommand(
+            name: "start-noservice",
+            description: @"Starts a server process without a backing symbol service client (useful for testing/pinging the daemon).",
+            needsIpcClient: false,
+            clientAction: (conf, _) =>
+            {
+                var symbolConfig = new SymbolConfig(string.Empty, new Uri("file://xyz"));
+                var daemonConfig = CreateDaemonConfig(conf);
+                var vsoClientTask = TaskSourceSlim.Create<ISymbolClient>();
+                vsoClientTask.SetException(new NotSupportedException());
+                using (var daemon = new SymbolDaemon(conf.Config.Parser, daemonConfig, symbolConfig, vsoClientTask.Task))
+                {
+                    daemon.Start();
+                    daemon.Completion.GetAwaiter().GetResult();

```

Comment:
Add a comment here saying that the service process pip execution will be blocked here while SymbolDaemon is busy with responding to IPC pip requests.

Reply:
Added the comment in the 'start' command since it's a more appropriate place.

---
In reply to: [314217639](https://github.com/microsoft/BuildXL/pull/727#discussion_r314217639) [](ancestors = 314217639)